### PR TITLE
Pf cluster validation

### PR DESCRIPTION
**Pull request by akanugan**
**PR description:**

Added an analyzer to get  PFCluster summed energies ratio plots and some profile histograms. 

**PR validation:**

Added an analyzer to get PFCluster summed energies plots and some profile histograms. runTheMatrix test workflows. The plots produced locally good enough. Plugin, interface and cfg files are created. Finally, modified macro and .html files to get the final set of plots.
Added plots are in the last 4 rows in the below link
http://hep.baylor.edu/cmshcal/akanugan/PFClusterValidation3/10_6_0_pre4_vs_10_6_0_pre4_SinglePi/

@mariadalfonso 
@abdoulline
 @bendavid
@arapyan

### CODE REVIEW

**Comment by akanugan** [comment]
@cmsbuild please test

**Comment by civanch** [comment]
please test

**Comment by jfernan2** [comment]
please test

**Comment by akanugan** [comment]
please test

**Comment by akanugan** [comment]
I have added a config file which was missing in the repo.
Now, the above matrix tests are tested and all of them passed.
I think it will pass the Jenkins test now.

**Comment by jfernan2** [comment]
please test

File: Validation/RecoParticleFlow/plugins/PFClusterValidation.cc [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+      consumes<reco::PFClusterCollection>(conf.getUntrackedParameter<edm::InputTag>("pflowClusterECAL"));
+  PFClusterHCALTok_ =
+      consumes<reco::PFClusterCollection>(conf.getUntrackedParameter<edm::InputTag>("pflowClusterHCAL"));
+  PFClusterHOTok_ = consumes<reco::PFClusterCollection>(conf.getUntrackedParameter<edm::InputTag>("pflowClusterHO"));
+  PFClusterHFTok_ = consumes<reco::PFClusterCollection>(
+      conf.getUntrackedParameter<edm::InputTag>("pflowClusterHF"));  // cms.InputTag("particleFlowClusterECAL");
+
+  outputFile_ = conf.getUntrackedParameter<std::string>("outputFile", "myfile.root");
+  mc_ = conf.getUntrackedParameter<std::string>("mc", "yes");
+
```
Comment (jfernan2) [inline Validation/RecoParticleFlow/plugins/PFClusterValidation.cc]: @akanugan why do you use here a string instead of a bool?

Comment (akanugan) [reply Validation/RecoParticleFlow/plugins/PFClusterValidation.cc]: @jfernan2 Hi, I can make that change but the same string was used in another module which I carried forward. If it's necessary, I can change it.

Comment (hatakeyamak) [reply Validation/RecoParticleFlow/plugins/PFClusterValidation.cc]: I think @akanugan meant 
https://cmssdt.cern.ch/lxr/source/Validation/CaloTowers/src/CaloTowersValidation.cc
Well, maybe we can fix this as well, i.e. use bool.
I think
@mariadalfonso
@abdoulline
@arapyan
will be happy if @akanugan does that?

**Comment by abdoulline** [comment]
Yes, Kenich. This string "yes" comes from old (dark) times...

File: Validation/RecoParticleFlow/plugins/PFClusterValidation.cc (hunk 1 above)
Comment (akanugan) [reply Validation/RecoParticleFlow/plugins/PFClusterValidation.cc]: The change has been made and also it passed the tests.

**Comment by jfernan2** [comment]
please test

**Comment by civanch** [comment]
+1

**Comment by jfernan2** [comment]
+1

**Comment by fabiocos** [comment]
@smuzaffar @jfernan2 in our guidelines we state "Do not split plugins into header and source files. If you do split them, keep the header files in the .../plugins directory." Here we do split the new PFClusterValidation, as the code is relatively long (not terribly but) I think this could be anyway acceptable.

**Comment by fabiocos** [comment]
+1

**Pull request merged**